### PR TITLE
Issue #1198: Add documentation for OOPs (Java Domain)

### LINKED ISSUE #1198: Java 2.2: OPPs Principal
<hr>

## Description 📜

Add audio, documentation, and video regarding the **OPPs Principal**.

<hr>

## Domain of Contribution 📊

- [x] Java

<hr>

## Location of File to be added

The files should be added inside the `OPPs Principal` folder which is inside the `Fundamental` present in the  `Java` main folder.  The documentation file can be in the `.md`/`.py`/`.ipynb` format. Audio should be in `.mp3` format and to be submitted through a Google drive link. Video can be in any format and should be submitted through a Google Drive link.

 **Note:**
 - You are required to comment with `\assign CONTENT_TYPE`, so that other content types can be created at the same time.
 - Failing to comment in the above manner will lead to the removal of assignment from the issue even if the bot assigns.
 - You are required to do a PR within 7 days otherwise you will be unassigned and others will be assigned.
 - Issues will be assigned on a first come first serve basis.

### PR DESCRIPTION
**Pull request by amitx007**
<hr>

## Description 📜
I have included all the information about the OOP principal.

Partially resolves #1198

<hr>

## Type of change 📝

- [ ] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [ ] Java_Domain
<hr>

## Checklist ✅

- [ ] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [ ] I have performed a self-review of my own code or work.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My changes generates no new warnings.
- [ ] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸
Screenshot 2021-09-23 105550
<hr>

### CODE REVIEW

**Comment by AkMo3** [comment]
@sakshamlamba109 Please find some time and complete the initial review.

File: Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+## Contents
+1. Introduction to class and object 
+2. What is a class ?
+3. What is an object?
+4. Difference between Classes and objects in Java
+5. Memory allocation 
+ 
+-------------
+## Introduction to class and object 
+- Classes and objects are one of the most fundametal things of many Java application and frameworks.
```
Comment (sakshamlamba109) [inline Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md:12]: grammatical error

change it to `fundamental units`

**Review by sakshamlamba109 — CHANGES_REQUESTED** [review]
@AkMo3 , I have made the changes

File: Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md (hunk 1 above)
Comment (amitx007) [reply Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md:12]: okay i will change it . Is there anything else

File: Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+- **Syntax** of defining object : 
+<!-- Syntax of creating object -->
+![object](https://github.com/amitx007/winter-of-contributing/blob/Java/Java/Fundamentals/Assets/objectsyntax.jpg)
+
+- Once we have created an object, then to access these class members we will use a special operator that is **Dot operator**.
+- **Syntax** of using dot operator:
+ <!-- Dot operator  -->
+![dot operator](https://github.com/amitx007/winter-of-contributing/blob/Java/Java/Fundamentals/Assets/dotoperator.jpg)
+----------
+##Difference between Classes and Objects in Java
```
Comment (AkMo3) [inline Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md:62]: Space after `##`

File: Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+|Class is created using keyword **class**                     |But object can be created in many ways and one of them is using **new**          keyword| 
+|Classes do not occupy memory location                         |But object occupy memory location  |
+|Classes can not be manipulated as its not available in memory |But we can manipulate objects|
+
+-----------
+## Memory Allocation
+- Memory Allocation is a process by which computer programs are assigned with some memory space .
+- As we know that object's address are stored in the reference variable of type class. Then to which place this address refer to ? Its the address of a space taken by the object in heap memory (which we will talk in this section) and that address is stored in reference variable. Can we print this address? **NO**, we can't print the address of the object but we can print the identity hashcode of that object by writing 
+```Java
+    object.toString();
```
Comment (AkMo3) [inline Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md:77]: I think it is `object.hashCode()`. See [Object Class JavaDocs](https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#hashCode())

**Review by AkMo3 — CHANGES_REQUESTED** [review]
Points:

File: Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+- **Behaviour** is the action or the work or the task the object performs. Here human is able to talk, talking is action or we can say behaviour. Similarly barking is a behaviour of dog. For better understanding you can look at this image 
+<!-- Image stating the difference between states and behaviour -->
+![state and behaviour](https://github.com/amitx007/winter-of-contributing/blob/Java/Java/Fundamentals/Assets/IMG1.jpg)
+- Now let us talk about classes. In layman term **Classes** are nothing but the logical description of how an object will behave and what are the states the object is going to have.
+- So let us take another **Example**. When you are creating a house, what is the first thing you do ? Yes you do the planning and make a sketch of it and if we make this sketch  with proper measurements and all, then it will be a **Blueprint**. So Blueprint will contain all the details and measurements of the floor ,ceiling and how it will be designed .
+<!-- Image of a house (blueprint) -->
+![class as a blueprint](https://github.com/amitx007/winter-of-contributing/blob/Java/Java/Fundamentals/Assets/IMG2.jpg)
```
Comment (AkMo3) [inline Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md:28]: You can use relative markdown as well. Change the URL to `../Assets/IMG2.jpg`.

**Review by AkMo3 — CHANGES_REQUESTED** [review]
Do similar changes at all places

File: Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md (hunk 4 above)
Comment (amitx007) [reply Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md]: everything done but couldn't understand what u mean by points:?

File: Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md (hunk 2 above)
Comment (amitx007) [reply Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md]: Done

File: Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md (hunk 3 above)
Comment (amitx007) [reply Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md]: Done

File: Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md (hunk 4 above)
Comment (amitx007) [reply Java/Fundamentals/OOPs Principals/OOPs_Principals_documentation.md]: @AkMo3  can u tell me abt the point that u have mentioned i couldn't understand
